src/js/ccfw-frontend.js

```diff
@@ -245,3 +249,59 @@ import { CCFW } from './ccfw-gtm';
         CCFW.manageAll(CCFW.storage.allowed.get(), 'init', true)
     });
 })(jQuery);
+
+function clearOurCookies(allowList) {
+    // Function to clear our cookies if consent withdrawn
+    if (!allowList.includes("ua")) {
+        //Google analytics
+        killCookieAndRelated("_ga");
+        killCookie("_gid");
+        killCookieAndRelated("_gat");
+    }
+    if (!allowList.includes("html")) {
+        //Facebook cookies
+        killCookie("fr");
+        killCookie("tr");
+        killCookie("_fbc");
+        killCookie("_fbp");
+        killCookie("PSUK_source");
+    }
+    if (!allowList.includes("gclidw")) {
+        //Google conversion linker cookies
+        killCookie("_gcl_au");
+        killCookie("_gcl_dc");
+        killCookie("_gcl_aw");
+    }
+}
+
+function killCookieAndRelated(name) {
+    //function for killing all cookies which start with <name>
+    // e.g. _ga will kill of _ga and _ga_123ABC
+    killCookie(name);
+    const cookies = document.cookie.split(";"); // array of cookies
+    for (var i = 0; i < cookies.length; i++) {
+        let cookie = cookies[i].trim();
+        if (!cookie) continue;
+        let eqPos = cookie.indexOf("=");
+        let fullname = eqPos > -1 ? cookie.substr(0, eqPos) : cookie;
+        if (fullname.substring(0,name.length) == name) {
+            killCookie(fullname);
+        }
+    }
+}
+
+function killCookie(name) {
+    // kills cookies of name for our domains
+    document.cookie = name + "=; expires=Sun, 01 May 1707 00:00:00 UTC; path=/;";
+    document.cookie = name + "=; expires=Sun, 01 May 1707 00:00:00 UTC; path=/;domain=" + location.host; // e.g. magistrates.judiciary.uk
+    document.cookie = name + "=; expires=Sun, 01 May 1707 00:00:00 UTC; path=/;domain=." + location.host; // e.g. .magistrates.judiciary.uk
+    let domain = location.host.split(".");
+    if (domain.length >= 3) domain[0] = "";
+    domain = domain.join(".");
+    document.cookie = name + "=; expires=Sun, 01 May 1707 00:00:00 UTC; path=/;domain=" + domain; // e.g. .judiciary.uk
+
+    //finally a few items which might be unnecessary as these are mostly wordpress login for editors and are essential cookies
```

Comment:
Could we make this more dynamic. Check if the site is on multisite and clear any cookies using the network_site_url. You could even have it so loops thought the '.' in the domain name. Im just thinking if we leave this hardcoded it needs to be updated when we move Cloud Platforms